htslib/hts_defs.h

```diff
@@ -69,6 +69,18 @@ DEALINGS IN THE SOFTWARE.  */
 #define HTS_DEPRECATED(message)
 #endif
 
+// On mingw the "printf" format type doesn't work.  It needs "gnu_printf"
+// in order to check %lld and %z, otherwise it defaults to checking against
```

Comment:
Agreed, where we have data types that are int32_t or int64_t we should (and do) use PRId64 etc.  The HTS_FORMAT macro though is generic and we don't have any way to enforce programs that use htslib stick to only int64_t and not "long long", so the change is simply to make it as robust as it can be.

We probably ought to fix such cases in our own code though - eg samtools/bam_stat.c has ```long long``` all over the place and hence (correct, given the type) ```%lld``` too.  It should use uint64_t for most of those.